### DISCUSSION TITLE
How to configure Host key algorithm for libssh?

### DISCUSSION THREAD

**Discussion by laogaolao** [libssh2/libssh2#1288] [Q&A] [answered]
In ssh_config, we can set ### **HostKeyAlgorithms ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521** to make sure the host key algorithm. It works fine when I use the ssh or sftp command in the linux shell.

But when I use libcurl to upload the file, a lot unexpected algorithm will send from client when I use network sniffer tool. So is there possible to set the HostKeyAlgorithm for libssh?

The algorithm packet captured in server side:
<img width="907" alt="image" src="https://github.com/libssh2/libssh2/assets/15045198/d71602f4-04b4-4e3a-8cd9-e2ae80cf05b2">

From the picture we can see ssh-dss, ssh-rsa.... which is unexpected. Thanks.

**Comment by willco007** [comment, marked as answer]
Use `libssh2_session_method_pref` with the flag `LIBSSH2_METHOD_HOSTKEY`.

**Comment by dfandrich** [comment]
You're asking a libssh question on a libssh2 forum. They are completely
independent projects.

**Comment by laogaolao** [comment]
@dfandrich meet you again. Sorry, I first touch these area, since the libcurl on our system is based on libssh2, so I think maybe I can find some answers here.

**Comment by laogaolao** [comment]
@willco007 I checked the LIBSSH2_METHOD_HOSTKEY, it shows only support two algorithms, is that correct? Thanks
LIBSSH2_METHOD_HOSTKEY
Public key algorithms. Supported values:

ssh-dss
Based on the Digital Signature Standard (FIPS-186-2)

ssh-rsa
Based on PKCS#1 (RFC 3447)

**Reply by willco007** [reply]
No, look in `hostkey.c` for supported values.

**Comment by laogaolao** [comment]
@willco007 Thank you, this works fine

**Comment by willco007** [comment]
Closing as completed.
